TombEngine/Game/control/control.cpp

```diff
@@ -478,7 +478,23 @@ void InitializeScripting(int levelIndex, bool loadGame)
 void DeInitializeScripting(int levelIndex, GameStatus reason)
 {
 	g_GameScript->OnEnd(reason);
-	HandleAllGlobalEvents(EventType::End, (Activator)LaraItem->Index);
+	switch (reason)
+	{
+	case GameStatus::LoadGame:
+		HandleAllGlobalEvents(EventType::BeforeLoad, (Activator)LaraItem->Index);
+		break;
+	case GameStatus::ExitToTitle:
+		HandleAllGlobalEvents(EventType::ExitToTitle, (Activator)LaraItem->Index);
+		break;
+	case GameStatus::LaraDead:
+		HandleAllGlobalEvents(EventType::Death, (Activator)LaraItem->Index);
+		break;
+	case GameStatus::LevelComplete:
+		HandleAllGlobalEvents(EventType::End, (Activator)LaraItem->Index);
```

Comment:
Can we keep same names for all events, in correlation with `GameStatus`? E.g. if GameStatus is `LoadGame`, can event be also called `EventType:: LoadGame`, and so on for all other events?